### ISSUE TITLE
Issue with `arrow` and a specific syntax

### ISSUE THREAD

**Issue by gmarty** [lebab/lebab#259]
I think the following syntax causes a bug with the `arrow` transform:
```js
var a = function(b) {
  return {
    a: 1,
    b: 2,
  }[b];
};
```

More specifically the `[b]` after the `return` seems to be the cause of it.

**Comment by nene** [comment]
Thanks for reporting.

I see that the transform works without crashing, but results in invalid JavaScript:

```js
var a = b => {
  a: 1,
  b: 2,
}[a];
```

it really should be:

```js
var a = b => ({
  a: 1,
  b: 2,
}[a]);
```

Seems to be a bug in Recast library that handles turning the transformed AST back to code. It has a history of bugs related to the placement of parentheses.

**Comment by nene** [comment]
Recast has on open bug for a very similar-looking issue: https://github.com/benjamn/recast/issues/351

**Comment by nene** [comment]
Fixed in 3.0.2
